obspy/core/util/base.py

```diff
@@ -60,6 +59,30 @@
 NATIVE_BYTEORDER = _sys_is_le and '<' or '>'
 
 
+# apply simple memoization cache on load_entry_points
+# this function cannot go in decorator.py due to circular import issues
+def _load_entry_point_decorator(func):
+    """
+    Decorate pkg_resources' load_entry_point function to cache outputs.
+    """
+    cache = {}
+    expected_keys = ('dist', 'group', 'name')
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # make a str key from the inputs to func for hashing in cache
+        input_dict = inspect.getcallargs(func, *args, **kwargs)
+        hash_str = '/'.join([input_dict[x] for x in expected_keys])
+        if hash_str not in cache:
+            cache[hash_str] = func(*args, **kwargs)
+        return cache[hash_str]
+
+    return wrapper
+
+
+load_entry_point = _load_entry_point_decorator(load_entry_point)
```

Comment:
Hmm.. is this patching it at `pkg_resources` directly, actually? If so, it might affect other installed python packages.. and I'd vote for only patching it inside ObsPy (i.e. just replace `load_entry_point` inside obspy with a cached version)..